reproject/interpolation/core_celestial.py

```diff
@@ -23,7 +25,8 @@ def _reproject_celestial(array, wcs_in, wcs_out, shape_out, order=1):
     """
 
     # Make sure image is floating point
-    array = np.asarray(array, dtype=float)
+    if not np.issubdtype(array.dtype, np.float):
+        array = np.asarray(array, dtype=float)
```

Comment:
@MSeifert04 empirically, it does not, even though the numpy docs say it should.  Though, maybe it is because my array had the opposite order or endianness or something. @astrofrog, I suspect the same would happen with your approach.  I actually prefer the approach here, because it explicitly avoids any unnecessary contact with the data.